src/Esprima/Utils/ToJavascriptConverter.cs

```diff
@@ -0,0 +1,1580 @@
+        }
+
+        protected virtual void VisitRestElement(RestElement restElement)
+        {
+            Append("...");
+            Visit(restElement.Argument);
+        }
+
+        protected virtual void VisitProperty(Property property)
+        {
+            if (property.Key is MemberExpression || ExpressionNeedsBrackets(property.Key))
+            {
+                Append("[");
+            }
+            if (ExpressionNeedsBrackets(property.Key))
+            {
+                Append("(");
+            }
+            Visit(property.Key);
+            if (ExpressionNeedsBrackets(property.Key))
+            {
+                Append(")");
+            }
+            if (property.Key is MemberExpression || ExpressionNeedsBrackets(property.Key))
+            {
+                Append("]");
+            }
+            if (property.Key is Identifier keyI && property.Value is Identifier valueI && keyI.Name == valueI.Name)
+            { }
+            else
+            {
+                AppendBeautificationSpace();
+                Append(":");
+                AppendBeautificationSpace();
+                if (property.Value is not ObjectPattern && ExpressionNeedsBrackets(property.Value))
+                {
+                    Append("(");
+                }
+                Visit(property.Value);
+                if (property.Value is not ObjectPattern && ExpressionNeedsBrackets(property.Value))
+                {
+                    Append(")");
+                }
+            }
+        }
+
+        protected virtual void VisitPropertyDefinition(PropertyDefinition propertyDefinition)
+        {
+            if (propertyDefinition.Static)
+            {
+                Append("static ");
+            }
+            if (propertyDefinition.Key is MemberExpression || ExpressionNeedsBrackets(propertyDefinition.Key))
+            {
+                Append("[");
+            }
+            if (ExpressionNeedsBrackets(propertyDefinition.Key))
+            {
+                Append("(");
+            }
+            Visit(propertyDefinition.Key);
+            if (ExpressionNeedsBrackets(propertyDefinition.Key))
+            {
+                Append(")");
+            }
+            if (propertyDefinition.Key is MemberExpression || ExpressionNeedsBrackets(propertyDefinition.Key))
+            {
+                Append("]");
+            }
+            if (propertyDefinition.Value != null)
+            {
+                Append("=");
+                Visit(propertyDefinition.Value);
+            }
+            Append(";");
+        }
+
+        protected virtual void VisitAwaitExpression(AwaitExpression awaitExpression)
+        {
+            Append("await ");
+            Visit(awaitExpression.Argument);
+        }
+
+        protected virtual void VisitConditionalExpression(ConditionalExpression conditionalExpression)
+        {
+            if (conditionalExpression.Test is AssignmentExpression)

```

Comment:
What about e.g. `var v = (a, b, c) ? d : e`? Looks like this problem is more widespread than it seems at first glance...